src/move_basic.cpp

```diff
@@ -440,6 +500,11 @@ bool MoveBasic::moveLinear(double requestedDistance)
              continue;
         }
 
+        if (obstacleDetected) {
+            abortGoal("Obstacle detected");
```

Comment:
I am not sure that aborting is the right call. It should probably just stop moving, and abort on timeout.